.github/workflows/compile.yml

```diff
@@ -19,13 +19,23 @@ jobs:
         sudo mkdir -p /usr/local
         curl --location https://github.com/rems-project/sail/releases/download/0.18-linux-binary/sail.tar.gz | sudo tar xvz --directory=/usr/local --strip-components=1
     - name: Build and test simulators
-      run: test/run_tests.sh
+      run: |
+        # Ninja is used because the CMake Makefile generator doesn't
+        # build top-level targets in parallel unfortunately.
+        cmake -S . -B build -GNinja -DCMAKE_BUILD_TYPE=RelWithDebInfo
+        # By default only the rv32d and rv64d emulators are build,
+        # but we want to build more targets here to ensure they
+        # can at least build without errors.
+        ninja -C build all riscv_sim_rv32d_rvfi generated_smt_rv64f generated_docs_rv64d
+        # These targets do not work with Sail 0.18: generated_rmem_rv32d_rmem generated_coq_rv64f_coq
+        ctest --test-dir build --output-junit tests.xml
     - name: Upload test results
       if: always()
       uses: actions/upload-artifact@v4
       with:
         name: tests.xml
-        path: test/tests.xml
+        path: build/tests.xml
+        if-no-files-found: error
```

Comment:
With the if: always() this will give cascading errors, but that's probably ok. Really what we want is "if we ran the test suite then upload test results regardless of exit code for ctest, but still fail the build if it had failures".

Reply:
Yeah I mainly use Gitlab CI so I'm not really sure how to do that, sorry! I guess 2 errors when you expect 1 is not too bad.